surefire-api/src/main/java/org/apache/maven/surefire/api/util/TempFileManager.java

```diff
@@ -0,0 +1,242 @@
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import java.io.File;
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicInteger;
+
+/**
+ * Management of temporary files in surefire with support for sub directories of the system's directory
+ * of temporary files.<br>
+ * The {@link File#createTempFile(String, String)} API creates rather meaningless file names and
+ * only in the system's temp directory.<br>
+ * This class creates temp files from a prefix, a unique date/timestamp, a short file id and suffix.
+ * It also helps you organize temporary files into sub-directories,
+ * and thus avoid bloating the temp directory root.<br>
+ * Apart from that, this class works in much the same way as {@link File#createTempFile(String, String)}
+ * and {@link File#deleteOnExit()}, and can be used as a drop-in replacement.
+ *
+ * @author Markus Spann
+ */
+public final class TempFileManager
+{
+
+    private static final Map<File, TempFileManager> INSTANCES  = new LinkedHashMap<>();
+    /** An id unique across all file managers used as part of the temporary file's base name. */
+    private static final AtomicInteger              FILE_ID    = new AtomicInteger( 1 );
+    private static final String                     SUFFIX_TMP = ".tmp";
+
+    private static Thread                           shutdownHook;
+
+    /** The temporary directory or sub-directory under which temporary files are created. */
+    private final File                              tempDir;
+    /** The fixed base name used between prefix and suffix of temporary files created by this class. */
+    private final String                            baseName;
+
+    /** List of successfully created temporary files. */
+    private final List<File>                        tempFiles;
+
+    /** Temporary files are delete on JVM exit if true. */
+    private boolean                                 deleteOnExit;
+
+    private TempFileManager( File tempDir )
+    {
+        this.tempDir = tempDir;
+        this.baseName = DateTimeFormatter.ofPattern( "yyyyMMddHHmmssSSS" ).format( LocalDateTime.now() );
```

Comment:
Use basic ISO 8601.

Reply:
`DateTimeFormatter` has no constant for a date/timestamp with milliseconds.